modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/model/TrafficSourceDetails.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.analytics.reports.web.internal.model;
+
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.util.ListUtil;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * @author David Arques
+ */
+public class TrafficSourceDetails {
+
+	public TrafficSourceDetails() {
+	}
+
+	public TrafficSourceDetails(
+		List<ReferringURL> referringDomains,
```

Comment:
I think that TrafficSourceDetails is a too generic name if we only keep ReferringURL in it.

Reply:
Well, that was on purpose, here It'll be added the properties for the rest of traffic sources details. Like `referringSocialMedia`.

Reply:
I think this is not very consistent, to have the countrySearchKeywords separate from the TrafficDetails, but try to mix social and referral.

Chose one approach, and keep everything the same way

Reply:
Agreed, I'll try to come up with a more consistent approach